fabtests/component/sock_test.c

```diff
@@ -0,0 +1,253 @@
+		if (new_sock < 0) {
+			printf("error during server init\n");
+			return -1;
+		}
+
+		(*clients)[i] = new_sock;
+		
+	}
+	return 0;
+}
+
+int client_connect(int connections, struct sockaddr_storage server_addr, int sock, int **clients)
+{
+	int i, ret, new_sock;
+	int optval = 1;
+
+	*clients = calloc(connections, sizeof(int));
+
+
+	for (i = 0; i < connections; i++) {
+
+		new_sock = socket(server_addr.ss_family, SOCK_STREAM, 0);
+		if (new_sock < 0)
+			return -1;
+
+		ret = setsockopt(new_sock, SOL_SOCKET, SO_REUSEADDR, (char *) &optval, sizeof(optval));
+		if (ret) {
+			printf("error setting socket options\n");
+			return ret;
+		}
+
+		ret = connect(new_sock, (struct sockaddr *)&server_addr, server_addr_len);
+		if (ret)
+			return ret;
+
+		(*clients)[i] = new_sock;
+	}
+
+	return ret;
+}
+
+double time_select(int connections, int server_sock, int *clients) 
+{
+	fd_set select_set;
+	int i, max_sock;
+	struct timeval timeout;
+	struct timespec start;
+	struct timespec stop;
+
+	timeout.tv_sec = 0;
+	timeout.tv_usec = 0;
+
+	FD_ZERO(&select_set);
+	FD_SET(server_sock, &select_set);
+
+	max_sock = server_sock;
+
+	for (i = 0; i < connections; i++) {
+		FD_SET(clients[i], &select_set);
+
+		if (clients[i] > max_sock)
+			max_sock = clients[i];
+
+	}
+
+	clock_gettime(CLOCK_MONOTONIC, &start);
+	select(max_sock + 1, &select_set, NULL, NULL, &timeout);
+	clock_gettime(CLOCK_MONOTONIC, &stop);
+
+	return (stop.tv_nsec + stop.tv_sec * 1000000000) - (start.tv_nsec + start.tv_sec * 1000000000);
+}
+
+double time_poll(int connections, int server_sock, int *clients)
+{
+	struct pollfd *poll_set;
+	int i;
+	struct timespec start;
+	struct timespec stop;
+
+	poll_set = malloc(sizeof(struct pollfd) * connections);
+
+	for (i = 0; i < connections; i++) {
+		poll_set[i].fd = clients[i];
+	}
+
+	clock_gettime(CLOCK_MONOTONIC, &start);
+	poll(poll_set, POLLIN, 0);
+	clock_gettime(CLOCK_MONOTONIC, &stop);
+
+	return (stop.tv_nsec + stop.tv_sec * 1000000000) - (start.tv_nsec + start.tv_sec * 1000000000);
+}
+
+double time_epoll(int connections, int server_sock, int *clients)
+{
+	struct epoll_event *events;
+	int efpd = epoll_create1(0);
+	int i, ret;
+	struct timespec start;
+	struct timespec stop;
+
+	events = malloc(sizeof(struct epoll_event) * connections);
+
+	for (i = 0; i < connections; i++) {
+
+		events[i].events = EPOLLIN;
+    	events[i].data.fd = clients[i];
+
+		ret = epoll_ctl(efpd, EPOLL_CTL_ADD, clients[i], &events[i]);
+		if (ret)
+			return ret;
+	}
+
+	clock_gettime(CLOCK_MONOTONIC, &start);
+	epoll_wait(EPOLLIN, events, connections, 0);
+	clock_gettime(CLOCK_MONOTONIC, &stop);
+	return (stop.tv_nsec + stop.tv_sec * 1000000000) - (start.tv_nsec + start.tv_sec * 1000000000);
+
+}	
+
+int main(int argc, char **argv)
+{
+	extern char *optarg;
+	int c, ret, connections, sock, i;
+	char *server_address, *mode;
+	struct sockaddr_storage server_addr;
+	int *clients;
+	double elapsed;
+
+	while ((c = getopt(argc, argv, "n:s:m:")) != -1) {

```

Comment:
Try to reuse the existing fabtests options, so a user doesn't need to learn new options just for this test.  In this case, use ADDR_OPTS.